src/iface/interface.rs

```diff
@@ -1082,24 +1082,16 @@ impl<'a> Interface<'a> {
             let mut respond = |inner: &mut InterfaceInner, response: IpPacket| {
                 neighbor_addr = Some(response.ip_repr().dst_addr());
                 match device.transmit().ok_or(Error::Exhausted) {
-                    Ok(_t) => {
+                    Ok(t) => {
                         #[cfg(feature = "proto-sixlowpan-fragmentation")]
-                        if let Err(e) = inner.dispatch_ip(_t, response, Some(_out_packets)) {
-                            net_debug!("failed to dispatch IP: {}", e);
-                            return Err(e);
-                        }
+                        inner.dispatch_ip(t, response, Some(_out_packets))?;
 
                         #[cfg(not(feature = "proto-sixlowpan-fragmentation"))]
-                        if let Err(e) = inner.dispatch_ip(_t, response, None) {
-                            net_debug!("failed to dispatch IP: {}", e);
-                            return Err(e);
-                        }
+                        inner.dispatch_ip(t, response, None)?;
+
                         emitted_any = true;
                     }
-                    Err(e) => {
-                        net_debug!("failed to transmit IP: {}", e);
```

Comment:
On first inspection, it looks like this transmit-specific debug message is being lost. If any of the other code paths in this closure return `Unaddressable`, it won’t be possible to distinguish between them.

Reply:
Totally right! I replaced the `unwrap_or` with an `unwrap_or_else` and log it there that there was a device exhausted error.